website/docs/plugin/framework/migrating/data-sources/index.mdx

```diff
@@ -0,0 +1,177 @@
+---
+page_title: 'Data Sources: Migrating from SDKv2 to the Framework'
+description: >-
+  Migrate a data source from SDKv2 to the plugin Framework.
+---
+
+# Data Sources
+
+Data sources let Terraform reference external data. Unlike resources, Terraform does not create, update, or delete
+data sources, and makes no attempt to modify the underlying API. Data Sources are a read-only resource type, so they
+only implement a subset of the operations that resources do. Refer to [Data Sources](/plugin/framework/data-sources)
+in the Framework documentation for details.
+
+This page explains how to migrate a data source from SDKv2 to the plugin Framework.
+
+## SDKv2
+
+In SDKv2, data sources are defined by the `DataSourcesMap` field on the `schema.Provider` struct, which maps data source
+names (strings) to their schema. The `schema.Resource` struct is used for both resources and data sources.
+
+The following example shows a typical implementation.
+
+```go
+func New() *schema.Provider {
+    return &schema.Provider{
+        DataSourcesMap:   map[string]*schema.Resource{}
+        /* ... */
+},
+```
+
+In SDKv2, you define both resources and data sources with `schema.Resource` structs. The following example shows a
+resource struct. For clarity, the example omits fields that are not available for data sources.
+
+```go
+schema.Resource {
+    Schema:               map[string]*schema.Schema,
+    Read:                 ReadFunc,
+    ReadContext:          ReadContextFunc,
+    ReadWithoutTimeout:   ReadContextFunc,
+    DeprecationMessage:   string,
+    Timeouts:             *ResourceTimeout,
+    Description:          string,
+}
+```
+
+## Framework
+
+In the Framework, you define data sources adding them to your provider's `GetDataSources` function.
+
+The `GetDataSources` function on your `provider.Provider` returns a map from the data source name (string) to a type
+that implements the `DataSourceType` interface for each data source your provider supports.
+
+The following code shows how you add a data source to your provider with the Framework.
+
+```go
+func (p *provider) GetDataSources(ctx context.Context) (map[string]provider.DataSourceType, diag.Diagnostics) {
+    return map[string]provider.DataSourceType{}, nil
+}
```

Comment:
```suggestion
func (p *provider) GetDataSources(ctx context.Context) (map[string]provider.DataSourceType, diag.Diagnostics) {
    return map[string]provider.DataSourceType{}, nil
}
```

Since this example is supposed to show how to add a data source, should this include the map itself? Something like:

```
func (p *provider) GetDataSources(ctx context.Context) (map[string]provider.DataSourceType, diag.Diagnostics) {
    return map[string]provider.DataSourceType{
        "data_source_example": dataSourceTypeExample{},
    }, nil
}
```

Reply:
I've added the usual placeholder (`/* ... */`) as this corresponds with the generic version we show for the Framework.